.github/workflows/main.yaml

```diff
@@ -0,0 +1,351 @@
+
+  gather-artifacts:
+    runs-on: ubuntu-16.04
+    needs: [build-kernel, build-qemu, build-qemu-virtiofsd, build-image, build-firecracker, build-kata-components]
+    steps:
+      - uses: actions/checkout@v1
+      - name: get-artifacts
+        uses: actions/download-artifact@master
+        with:
+          name: kata-artifacts
+      - name: colate-artifacts
+        run: |
+          $GITHUB_WORKSPACE/.github/workflows/gather-artifacts.sh
+      - name: store-artifacts
+        uses: actions/upload-artifact@master
+        with:
+          name: release-candidate
+          path: kata-static.tar.xz
+
+  kata-deploy:
+    needs: gather-artifacts
+    runs-on: ubuntu-latest
+    steps:
+      - name: get-artifacts
+        uses: actions/download-artifact@master
+        with:
+          name: release-candidate
+      - name: build-and-push-kata-deploy-ci
+        run: |
+          tag=`echo $GITHUB_REF | cut -d/ -f3-`
+          git clone https://github.com/kata-containers/packaging
+          pushd packaging
+          git checkout $tag
+          pkg_sha=$(git rev-parse HEAD)
+          popd
+          mv release-candidate/kata-static.tar.xz ./packaging/kata-deploy/kata-static.tar.xz
+          docker build --build-arg KATA_ARTIFACTS=kata-static.tar.xz -t katadocker/kata-deploy-ci:$pkg_sha ./packaging/kata-deploy
+          docker login -u ${{ secrets.DOCKER_USERNAME }} -p ${{ secrets.DOCKER_PASSWORD }}
+          docker push katadocker/kata-deploy-ci:$pkg_sha
+          echo ::set-env name=PKG_SHA::$pkg_sha
+          echo ::set-env name=TAG::$tag
+      - name: test-kata-deploy-ci-in-aks
+        uses: ./packaging/kata-deploy/action
+        with:
+          packaging-sha: env.PKG_SHA
+        env:
+          PKG_SHA: ${{ env.PKG_SHA }}
+          AZ_APPID: ${{ secrets.AZ_APPID }}
+          AZ_PASSWORD: ${{ secrets.AZ_PASSWORD }}
+          AZ_SUBSCRIPTION_ID: ${{ secrets.AZ_SUBSCRIPTION_ID }}
+          AZ_TENANT_ID: ${{ secrets.AZ_TENANT_ID }}
+      - name: push-tarball
+        run: |
+          # tag the container image we created and push to DockerHub
+          docker tag katadocker/kata-deploy-ci:${{ env.PKG_SHA }} katadocker/kata-deploy:${{ env.TAG }}
+          docker push katadocker/kata-deploy:${{ env.TAG }}
+
+  upload-static-tarball:
+    needs: kata-deploy
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code
+        uses: actions/checkout@master
+      - name: install-go
+        uses: actions/setup-go@v1
+        with:
+          go-version: '1.13.x'
+      - name: get-artifacts
+        uses: actions/download-artifact@master
+        with:
+          name: release-candidate
+      - name: install hub
+        run: |
+          git clone \
+             --config transfer.fsckobjects=false \
+             --config receive.fsckobjects=false \
+             --config fetch.fsckobjects=false \
+             https://github.com/github/hub.git
+          cd hub
+          sudo make install prefix=/usr/local
```

Comment:
Would it be better to use releases?
```
        wget -q -O- \
          https://github.com/github/hub/releases/download/v$HUB_VER/hub-linux-amd64-$HUB_VER.tgz | \
          tar xz --strip-components=2 --wildcards '*/bin/hub' && sudo mv hub /bin/hub
```

Reply:
is there a way to get latest stable version, though?  @krsna1729 @amshinde ?

Reply:
@egernst from stackoverflow
```
HUB_VER=$(curl -s https://github.com/github/hub/releases/latest | grep -Po 'href="\K.*?(?=")' | awk -F '/' '{print substr($NF,2)}')
```

Reply:
@krsna1729 @egernst Using a stable release now. Decided to go with github api and `jq` to get the latest release.